source/NVDAObjects/UIA/__init__.py

```diff
@@ -2019,11 +2019,13 @@ def doAction(self,index=None):
 		if index==0:
 			if self.UIAInvokePattern:
 				self.UIAInvokePattern.Invoke()
+				return
 			elif self.UIATogglePattern:
 				self.UIATogglePattern.toggle()
+				return
 			elif self.UIASelectionItemPattern:
 				self.UIASelectionItemPattern.select()
-			return

```

Comment:
Why this change? I don't think there's a functional difference, but having elif + return is redundant. If you want to keep the early returns in each branch, I'd change the elifs to ifs.

Reply:
Previously it would raise NotImplementedError only if actionIndex was not 0. Now it also raises NotImplementedError if neither invoking, toggling or selecting is possible. At least, that was what I was trying to go for...

Reply:
Ah gotcha. Sorry, missed that subtlety. In that case, please change the elifs to ifs and we're good here.